File: examples/collections/rust/core/bytes.rs
use crate::core_array::*;
use crate::core_iterator::*;
use crate::core_list::*;
use crate::core_map::*;
use crate::core_sorted::*;
use crate::core_string::*;
use crate::unions::*;

pub fn iter__2(data: &Vec<u8>) -> BytesYield<'_> {
    return BytesYield { data: data, at: 0 };
}

#[derive(Clone, Debug, PartialEq)]
pub struct BytesYield<'s> {
    pub data: &'s Vec<u8>,
    pub at: i32,
}

pub fn next__2(p: &mut BytesYield<'_>) -> Union2<u8, Finished> {
    let mut b = p.data.get((p.at) as usize).copied();
    if b.is_none() {
        return Union2::<u8, Finished>::U2(finished());
    }
    p.at = p.at + 1;
    return Union2::<u8, Finished>::U1(emitted(b.unwrap()));
}

File: examples/throw-and-release/kotlin/core/bytes.kt
package salvo.core.bytes

import salvo.*
import salvo.core.iterator.*
import salvo.core.list.*
import salvo.core.map.*
import salvo.core.sorted.*
import salvo.core.string.*

fun iter__2(data: salvo.SalvoBytes): BytesYield {
    return BytesYield(data = data, at = 0)
}

data class BytesYield(
    var data: salvo.SalvoBytes,
    var at: Int,
)

fun next__2(p: BytesYield): Union2<UByte, Finished> {
    val b = p.data.getOrNull(p.at)
    if (b == null) {
        return U2_2<UByte, Finished>(finished())
    }
    p.at = p.at + 1
    return U2_1<UByte, Finished>(emitted(b))
}
